cli/dcoscli/package/main.py

```diff
@@ -499,27 +499,33 @@ def _uninstall(package_name, remove_all, app_id, cli, app, skip_confirmation):
     :rtype: int
     """
 
+    package_manager = get_package_manager()
+
     # Don't gate CLI uninstalls.
     # Don't gate uninstalls if the user wants to skip confirmation.
-    if not cli and not skip_confirmation and not _confirm_uninstall(
-        package_name,
-        remove_all,
-        app_id
-    ):
-        return 0
+    if not cli and not skip_confirmation:
+        installed = package.installed_packages(
+            package_manager, None, package_name, cli_only=False)
+        installed_pkg = next(iter(installed), None)
+        if not installed_pkg or not installed_pkg.get('name'):
+            msg = "Package '{}' is not installed.".format(package_name)
+            emitter.publish(msg)
+            return 1
```

Comment:
Could we push this case down into `_confirm_uninstall()`?

Also, what is the intended behaviour if the package doesn't exist and we call `_uninstall()` with `cli=True` or `skip_confirmation=True`?

Reply:
The reason I've left it outside is because a missing package would result in an exit code error while a non confirmation would just exit the program successfully, so I would have to adapt the function with these new semantics (and also add the `package_name` parameter needed for the "Package not installed" error message), which felt too much to me.

If the confirmation is skipped and the package doesn't exist, there is an error returned by the underlying API (I think it's Cosmos)  that would be printed, eg `Package [foo] is not installed`. This was actually one of the things mentioned in the JIRA issue, we are currently asking confirmation even for package which doesn't exist, before failing saying they don't exist..